src/Symfony/Component/Console/Command/AlarmableCommandInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Input\InputInterface;
+
+/**
+ * Interface for commands that listen to SIGALRM signals.
+ */
+interface AlarmableCommandInterface
+{
+    /**
+     * The method will be called before the command is run and subsequently on each SIGALRM signal.
+     *
+     * @return int The alarm interval in seconds
+     */
+    public function getAlarmInterval(InputInterface $input): int;
```

Comment:
instead of adding this interface, what about adding Application::setAlarmInterval? A command would then have to call the method is execute and that'd make things simpler, don't you think?

Reply:
@nicolas-grekas Interesting idea, I'm guessing that the `handleAlarm` would then be replaced with the standard `SignalableCommandInterface`?

Reply:
absolutely

Reply:
@nicolas-grekas Please take a look at the new implementation https://github.com/symfony/symfony/pull/53508/commits/5c32d9c9e507580698a5a029a43eac743d6c37b6